File: scala-zio/src/main/scala/com/claude/bakeoff/config/MongoConfig.scala
package com.claude.bakeoff.config

import com.mongodb.client.{MongoClient, MongoClients}
import zio.*

object MongoConfig:
  val layer: ZLayer[Any, Nothing, MongoClient] = ZLayer.scoped {
    ZIO.acquireRelease(
      ZIO.succeed {
        val connectionString = sys.env.getOrElse("MONGODB_URI", 
          "mongodb://admin:password@mongodb:27017/bakeoff?authSource=admin")
        MongoClients.create(connectionString)
      }
    )(client => ZIO.succeed(client.close()))
  }

File: kotlin-ktor/src/main/kotlin/com/claude/bakeoff/config/MongoConfig.kt
package com.claude.bakeoff.config

import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients

object MongoConfig {
    
    fun createMongoClient(): MongoClient {
        val connectionString = System.getenv("MONGODB_URI") 
            ?: "mongodb://admin:password@mongodb:27017/bakeoff?authSource=admin"
        return MongoClients.create(connectionString)
    }
}